go-controller/pkg/cni/udn/primary_network.go

```diff
@@ -104,6 +104,29 @@ func (p *UserDefinedPrimaryNetwork) WaitForPrimaryAnnotationFn(namespace string,
 	}
 }
 
+func (p *UserDefinedPrimaryNetwork) Ensure(namespace string, annotations map[string]string, nadName string) error {
```

Comment:
could we re-use this method from the `WaitForPrimaryAnnotationFn` function ? 

Not sure it fits well ...

Reply:
We should, there is a common important bits here.